### PR TITLE
977/deprecate rinkeby

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Part of #977 

Disabling Rinkeby in the UI

This change is minimal and now the app respects the networks selected in the envvar `REACT_APP_SUPPORTED_CHAIN_IDS`

Full removal should be performed once the SDK removes rinkeby from the list of supported networks

  # To Test

1. Without connecting you wallet, observe the network selection
* It should not contain Rinkeby
2. Connect you wallet in a network other than rinkeby
* App should work as usual
3. Change the network in your wallet to rinkeby
* You should see the network unsupported error

### CODE REVIEW

File: src/custom/components/Header/NetworkSelector/NetworkSelectorMod.tsx [hunk 1]
```diff
@@ -323,16 +321,14 @@ export const getChainNameFromId = (id: string | number) => {
   return CHAIN_IDS_TO_NAMES[id as SupportedChainId] || ''
 }
 
-export const NETWORK_SELECTOR_CHAINS = [
+export const NETWORK_SELECTOR_CHAINS: SupportedChainId[] = SUPPORTED_CHAIN_IDS
```
Comment (alfetopito) [inline src/custom/components/Header/NetworkSelector/NetworkSelectorMod.tsx:324]: No more hardcoded networks, now it depends on what is actually enabled

File: src/custom/connection/index.ts [hunk 2]
```diff
@@ -12,19 +12,6 @@ import Fortmatic from 'fortmatic'
 import UNISWAP_LOGO_URL from 'assets/svg/logo.svg'
 import { RPC_URLS } from 'constants/networks'
 
-// Mod
-export function getSupportedChainIds(): number[] {
```
Comment (alfetopito) [inline src/custom/connection/index.ts:16]: Moved to a different file, didn't make sense to have it here IMO

File: src/custom/utils/blocknative.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ interface SDKError {
 
 export const sdk = !BLOCKNATIVE_API_KEY
   ? {}
-  : getSupportedChainIds().reduce<Record<number, BlocknativeSdk | null>>((acc, networkId) => {
+  : SUPPORTED_CHAIN_IDS.reduce<Record<number, BlocknativeSdk | null>>((acc, networkId) => {
```
Comment (alfetopito) [inline src/custom/utils/blocknative.ts:15]: The only place where this function was used

File: src/custom/utils/supportedChainId.ts [hunk 4]
```diff
@@ -1,18 +1,33 @@
...
-  return Boolean(SupportedChainId[chainId])
-}
-
 /**
  * Returns the input chain ID if chain is supported. If not, return undefined
  * @param chainId a chain ID, which will be returned if it is a supported chain ID
  */
-//  export function supportedChainId(chainId: number): number | undefined {
-export function supportedChainId(chainId?: number): SupportedChainId | undefined {
+export function supportedChainId(chainId: number | undefined): SupportedChainId | undefined {
```
Comment (alfetopito) [inline src/custom/utils/supportedChainId.ts:7]: Signature changes are just to match original Uni definition

File: src/custom/utils/supportedChainId.ts [hunk 5]
```diff
@@ -1,18 +1,33 @@
...
   return undefined
 }
+
+// Mod
+export function isSupportedChain(chainId?: number): chainId is SupportedChainId {
+  if (!chainId) {
+    return false
+  }
+
+  return typeof chainId === 'number' && chainId in SupportedChainId && SUPPORTED_CHAIN_IDS.includes(chainId)
```
Comment (alfetopito) [inline src/custom/utils/supportedChainId.ts:20]: This is where the magic happens. Besides the type definition, check also the networks enabled in the env var

**Comment by elena-zh** [comment]
Hey @alfetopito , great and sad changes (I'm a Rinkeby billionaire!) 

So that is why some issues from my side: 
1. Fees discount is still calculated then connected to Rinkeby

2. The app imports USDC token and shows it in the 'To' token (and its balance). 


However, no token should be imported, 'Select token' should be displayed in the To field
3. I'm not sure if it is related, but [BE request](https://barn.api.cow.fi/rinkeby/api/v1/account/0xFF714b8b0e2700303eC912BD40496C3997ceEa2b/orders/?limit=100&offset=0) is running when I'm conencted to Rinkeby

4. I'm not sure, but should we displaye ENS name of an account when connected to Rinkeby?

5. These changes are included into #932 PR, and are missing here: the message on yellow banner contains Gorly (it is absent in the current PR). Besides, Activity modal also contains a message, but here it is empty. 

Here: 


Thanks!

**Comment by alfetopito** [comment]
> Hey @alfetopito , great and sad changes (I'm a Rinkeby billionaire!)

💸 💸 💸 

> So that is why some issues from my side:
> 
>     1. Fees discount is still calculated then connected to Rinkeby

✅ 

>     2. The app imports USDC token and shows it in the 'To' token (and its balance).

✅ 

>     3. I'm not sure if it is related, but [BE request](https://barn.api.cow.fi/rinkeby/api/v1/account/0xFF714b8b0e2700303eC912BD40496C3997ceEa2b/orders/?limit=100&offset=0) is running when I'm conencted to Rinkeby

✅ 

>     4. I'm not sure, but should we displaye ENS name of an account when connected to Rinkeby?

I don't mind very much about this one. It's more about the connected wallet so I'm fine with it

>     5. These changes are included into [Account page updates #932](https://github.com/cowprotocol/cowswap/pull/932) PR, and are missing here: the message on yellow banner contains Gorly (it is absent in the current PR). Besides, Activity modal also contains a message, but here it is empty.

Will be included when that PR is merged to develop

**Comment by elena-zh** [comment]
@alfetopito , all changes LGTM. 
But I have a doubt in regard to the last point. Look: I see the message when connected to another but Rinkeby unsupported network. 
other chains

And see nothing when connected to Rinkeby.
rinkeby

So I still assume that it is an issue related to this PR.

**Comment by alfetopito** [comment]
> @alfetopito , all changes LGTM. But I have a doubt in regard to the last point. Look: I see the message when connected to another but Rinkeby unsupported network. other chains
> 
> And see nothing when connected to Rinkeby. rinkeby
> 
> So I still assume that it is an issue related to this PR.

Thanks Elena!
Merged with latest from develop, which now include Nenad's changes and fixed the display for Rinkeby

**Review by elena-zh — APPROVED** [review]
@alfetopito , changes LGTM!
Just the last clarification: we are not going to show 'Gorli' in the list of supported networks on the pop-up. correct? 
image

**Comment by alfetopito** [comment]
> @alfetopito , changes LGTM! Just the last clarification: we are not going to show 'Gorli' in the list of supported networks on the pop-up. correct? image

It does now, the build was broken and updates weren't deployed
Screen Shot 2022-08-31 at 11 52 33

**Comment by elena-zh** [comment]
Looks good now!
I have just remembered that you removed a default token pair in the current PR.
I have just done another round of testing and noticed that it would be nice to remove this 0 when ETH/WETH token pair is selected. Could you please fix this?
Reproducible for cases when a user is not connected and when Insufficient balance error is displayed
image
image

**Review by nenadV91 — APPROVED** [review]
Codewise LGTM, but not sure if we wan't to release this yet.

**Review by anxolin — APPROVED** [review]

**Comment by anxolin** [comment]
@nenadV91 we want to release it. Why should we delay it? 

The sooner we do it, the sooner backend can remove the services (adds overhead to them).

**Comment by elena-zh** [comment]
@anxolin , my point is that we still do not have claim from locked GNO and convert COW from vCOW functionality in Gorli. 
image

So that is why I asked to postpone releasing this PR and wait till this functionality appears in Gorli.

**Comment by anxolin** [comment]
@elena-zh i was under the impression we won't have this in Goerli, just to not add some work on a feature that was already deployed. However, you have a point, if in a version we break the claiming, its hard to test it if we don't have a testnet (we can't test in production every time)

I will check with the backend

**Comment by anxolin** [comment]
The backend deployed already the contracts, so I think is in @nenadV91 roof to double check the deployment is correct, and we have some test account we can use

**Pull request merged**